cmd/format.go

```diff
@@ -222,3 +243,186 @@ func normalizeLineEndings(content []byte) []byte {
 }
+
+func runSqlfluffFormatting(repoPath string, logger interface{}) error {
+	// Find all SQL files with their asset information
+	sqlFilesWithDialects, err := findSQLFilesWithDialects(repoPath)
+	if err != nil {
+		return errors2.Wrap(err, "failed to find SQL files")
+	}
+
+	if len(sqlFilesWithDialects) == 0 {
+		return nil // No SQL files to format
+	}
+
+	// Find the git repo
+	repoFinder := &git.RepoFinder{}
+	repo, err := repoFinder.Repo(repoPath)
+	if err != nil {
+		return errors2.Wrap(err, "failed to find git repository")
+	}
+
+	// Create sqlfluff runner
+	sqlfluffRunner := &python.SqlfluffRunner{
+		UvInstaller: &python.UvChecker{},
+	}
+
+	// Run sqlfluff on all SQL files
+	ctx := context.WithValue(context.Background(), executor.ContextLogger, logger)
+	return sqlfluffRunner.RunSqlfluffWithDialects(ctx, sqlFilesWithDialects, repo)
+}
+
+func runSqlfluffFormattingForSingleAsset(assetPath string, repo *git.Repo, logger interface{}) error {
+	// Parse the single asset to get its type
+	asset, err := DefaultPipelineBuilder.CreateAssetFromFile(assetPath, nil)
+	if err != nil {
+		return errors2.Wrap(err, "failed to parse asset")
+	}
+
+	if asset == nil {
+		return nil // Not a valid asset
+	}
+
+	// Detect dialect from asset type
+	dialect := python.DetectDialectFromAssetType(string(asset.Type))
+
+	// Convert to absolute path for now since we're working in a nested repo structure
+	absolutePath, err := filepath.Abs(assetPath)
+	if err != nil {
+		return errors2.Wrap(err, "failed to get absolute path")
+	}
+
+	sqlFileInfo := python.SQLFileInfo{
+		FilePath: absolutePath,
+		Dialect:  dialect,
+	}
+
+	// Create sqlfluff runner
+	sqlfluffRunner := &python.SqlfluffRunner{
+		UvInstaller: &python.UvChecker{},
+	}
+
+	// Run sqlfluff on the single file
+	ctx := context.WithValue(context.Background(), executor.ContextLogger, logger)
+	return sqlfluffRunner.RunSqlfluffWithDialects(ctx, []python.SQLFileInfo{sqlFileInfo}, repo)
+}
+
+func findSQLFilesWithDialects(repoPath string) ([]python.SQLFileInfo, error) {
+	// Check if we're formatting a single asset
+	if isPathReferencingAsset(repoPath) {
+		// Single asset - detect dialect from the asset itself
+		asset, err := DefaultPipelineBuilder.CreateAssetFromFile(repoPath, nil)
+		if err != nil {
+			return nil, errors2.Wrap(err, "failed to parse asset")
+		}
+
+		if asset != nil {
+			dialect := python.DetectDialectFromAssetType(string(asset.Type))
+			return []python.SQLFileInfo{{
+				FilePath: repoPath,
+				Dialect:  dialect,
+			}}, nil
+		}
+		return []python.SQLFileInfo{}, nil
+	}
+
+	// Full pipeline - analyze all SQL files and detect dialects in parallel
+	sqlPaths := path.GetAllPossibleAssetPaths(repoPath, assetsDirectoryNames, []string{".sql"})
+
+	if len(sqlPaths) == 0 {
+		return []python.SQLFileInfo{}, nil
+	}
+
+	// Use conc pool to process assets in parallel with max 30 goroutines
+	assetPool := pool.New().WithMaxGoroutines(30)
```

Comment:
Would be nice to have the concurrency configurable. Depending on the number of assets, a higher or lower number may offer better performance.